sktime/forecasting/naive.py

```diff
@@ -176,3 +193,20 @@ def _predict_last_window(self, fh, X=None, return_pred_int=False,
                 # get zero-based index by subtracting the minimum
                 fh_idx = fh.index_like(self.cutoff)
                 return y_pred[fh_idx]
+
+        elif self.strategy == "drift":
+            if self.window_length_ != 1:
+                # if NaNs exists they are counted as zero and jump
+                # in linear difference is splitted among
+                # the missing values
+                drift = np.mean(np.diff(np.nan_to_num(last_window)))
```

Comment:
I don't understand why we can just replace missing values with zeros? We need to compute the difference between the last value and the first value, no? Check out the formula here: https://otexts.com/fpp3/simple-methods.html

I think we want something like this: 
```python
slope = (last_window[last] - last_window[first]) / (last - first)  # formula for slope
fh_idx = fh.index_like(self.cutoff)  # forecasting horizon
y_pred = last_window[last] + fh_idx * slope  # linear extrapolation
return y_pred 
```
This seems to be all we need for the case with no missing values. If we have missing values, and if the first or last value is a missing value, we need to take the first or last non-missing value. So we could do something like this:
```python
idx = np.argwhere(~np.isnan(last_window))  # get the index of non-missing values
first = idx[0]
last = idx[-1]
```
In the case where the last value is missing we still need to do one more thing: we need to estimate the missing value(s) given the last non-missing value and slope: 
```
# last value is non-missing
if last == (len(last_window) - 1):
    last_value = last_window[last]
# last value is missing
else:
    last_value = last_window[last] + slope * len(last_window) - last - 1

y_pred = last_value + fh_idx * slope
```

Reply:
This does seem to be a lot of code for a simple method, there may be a better way to implement it. We could also just raise an Error if there are missing values.